src/ledgerSigner/index.js

```diff
@@ -42,30 +48,74 @@ class LedgerNanoS {
   }
 
   getEthAddress = async (accountIndex) => {
-    const accountPath = basePath + `/${accountIndex}`
+    const accountPath = baseEtherPath + `/${accountIndex}`
     const ethLedger = await this.getEtherLedger()
     const result = await ethLedger.getAddress(accountPath)
     return result.address
   }
 
-  getBalance = async (accountIndex) => {
-    const address = await this.getEthAddress(accountIndex)
-    const web3 = this.getWeb3()
-    const balance = await web3.eth.getBalance(address)
-    return new BN(balance)
+  getBtcAddresss = async (accountIndex) => {
+    const btcLedger = await this.getBtcLedger()
+    const accountPath = `${baseBtcPath}/${accountIndex}'/0/0`
+    const addr = await btcLedger.getWalletPublicKey(accountPath, false, true)
+    return addr
+  }
+
+  getBtcLedger = async () => {
+    if (!this.btcLedger) {
+      this.btcLedger = new BtcLedger(await this.getTransport())
+    }
+    return this.btcLedger
   }
 
-  deviceConnected = async () => {
+  syncAccountBaseOnCryptoType = async (cryptoType, accountIndex = 0) => {
+    let address, web3, balance
+    switch (cryptoType) {
+      case 'ethereum':
+        address = await this.getEthAddress(accountIndex)
+        web3 = this.getWeb3()
+        balance = await web3.eth.getBalance(address)
+        return {
+          [cryptoType]: {
+            [accountIndex]: {
+              address: address,
+              balance: balance
+            }
+          }
+        }
+      case 'dai':
+      // TODO: add supoort for dai
+        address = await this.getEthAddress(accountIndex)
```

Comment:
duplicated code, consider merge this into ethereum-based coins section

```js
case 'ethereum':
case 'dai':
  // load eth balance
case 'dai':
  // load dai balance
```

Reply:
Will do in next commit on Ledger ERC20 tx.